pkgs/by-name/co/colstr/package.nix

```
@@ -0,0 +1,44 @@
+{ lib
+, stdenv
+, fetchFromGitea
+, nix-update-script
+}:
+
+stdenv.mkDerivation rec {
+  pname = "colstr";
+  version = "1.0.0";
+
+  src = fetchFromGitea {
+    domain = "git.sleeping.town";
+    owner = "wonder";
+    repo = "colstr";
+    rev = version;
+    hash = "sha256-0V2S/yYu5L7qxkT4Zf18x9+cHoPMztFmgSywpxF8QqA=";
+  };
+
+  buildPhase = ''
+    runHook preBuild
+
+    $CXX main.cpp
```

From my experience on Bash oddities, I would prefix this with eval:

```suggestion
    eval $CXX main.cpp
```

Or better - since `c++`  is the standardized name of this compiler,

```suggestion
    eval ${stdenv.cc.targetPrefix}c++ main.cpp
```